plugin/storage/badger/samplingstore/storage.go

```diff
@@ -91,12 +91,12 @@ func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput
 			if err != nil {
 				return err
 			}
-			throughputs, err := decodeThroughtputValue(val)
+			throughputs, err := decodeThroughputValue(val)
 			if err != nil {
 				return err
 			}
 
-			if t.After(start) && (t.Before(end) || t.Equal(end)) {
+			if (t.After(start) || t.Equal(start)) && (t.Before(end) || t.Equal(end)) {
```

Comment:
Why is this the right change?

Reply:
i made this changes to handle the condition when timestamps sometimes becomes equal (you can see at the top left corner) .
![image](https://github.com/jaegertracing/jaeger/assets/122517264/28a40a42-0bd6-4872-8d64-71b5638b03be)

>Why is this the right change?

Because it tackles the issue at the root cause, and we don't have to rely on some arbitrary delay. It will also ensure that operations occurring exactly at the start time are included in the results.

Reply:
> you can see at the top left corner

I am not sure what you're pointing at - the two INFO logs? The timestamps there are truncated to full seconds, so impossible to say anything (since we know already the difference in milliseconds affects the behavior).

> Because it tackles the issue at the root cause

What is your evidence of that? This change modifies the semantics of the query, changing timestamp interval treatment from open-closed to closed-closed.

Reply:
> The timestamps there are truncated to full seconds

The timestamps are not truncated to the full second. It has a precision of microseconds, which means it includes fractions of a second up to six decimal places (624305 microseconds) 
The presence of digits after the decimal point in the seconds field indicates microsecond precision.
Truncation to the full second would remove those digits, leaving only 21:35:23

Reply:
Here is another example of a passing test 
![image](https://github.com/jaegertracing/jaeger/assets/122517264/25dfccd3-2d9b-429b-99d9-725dd2912e64)
there is as slight difference of 1 microsecond

Reply:
Sorry, I don't understand what you are showing with this screenshot

Reply:
i am trying to show that the test passes when there is some difference in the timestamps, and it fails when the timestamps are same.

Reply:
how does the screenshot illustrate that?

Reply:
sorry, the above screenshot doesn't shows full screen shot
here are full logs of the **passed test**
```
harsh@MSI:~/jaeger/plugin/storage/badger/samplingstore$ go test
badger 2024/01/15 01:54:04 INFO: All 0 tables opened in 0s
badger 2024/01/15 01:54:04 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 01:54:04 INFO: Set nextTxnTs to 0
badger 2024/01/15 01:54:04 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 01:54:04 INFO:
Level 0 [ ]: NumTables: 01. Size: 282 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
badger 2024/01/15 01:54:04 INFO: All 0 tables opened in 0s
badger 2024/01/15 01:54:04 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 01:54:04 INFO: Set nextTxnTs to 0
usec:  2024-01-15 01:54:04.181913 +0530 IST
start:  2024-01-15 01:54:04.181912 +0530 IST
badger 2024/01/15 01:54:04 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 01:54:04 INFO:
Level 0 [ ]: NumTables: 01. Size: 282 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
badger 2024/01/15 01:54:04 INFO: All 0 tables opened in 0s
badger 2024/01/15 01:54:04 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 01:54:04 INFO: Set nextTxnTs to 0
badger 2024/01/15 01:54:04 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 01:54:04 INFO:
Level 0 [ ]: NumTables: 01. Size: 278 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
badger 2024/01/15 01:54:04 INFO: All 0 tables opened in 0s
badger 2024/01/15 01:54:04 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 01:54:04 INFO: Set nextTxnTs to 0
badger 2024/01/15 01:54:04 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 01:54:04 INFO:
Level 0 [ ]: NumTables: 01. Size: 332 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
PASS
ok      github.com/jaegertracing/jaeger/plugin/storage/badger/samplingstore     0.496s
```

you can see in the logs that there is a difference of 1 microsecond and because of that difference test passed 
```
usec:  2024-01-15 01:54:04.181913 +0530 IST
start:  2024-01-15 01:54:04.181912 +0530 IST
```

Reply:
here are logs of failing test 
```
badger 2024/01/15 02:03:49 INFO: All 0 tables opened in 0s
badger 2024/01/15 02:03:49 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 02:03:49 INFO: Set nextTxnTs to 0
usec:  2024-01-15 02:03:49.674922 +0530 IST
start:  2024-01-15 02:03:49.674922 +0530 IST
badger 2024/01/15 02:03:49 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 02:03:49 INFO:
Level 0 [ ]: NumTables: 01. Size: 282 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
--- FAIL: TestGetThroughput (0.06s)
    storage_test.go:59:
                Error Trace:    /home/harsh/jaeger/plugin/storage/badger/samplingstore/storage_test.go:59
                                                        /home/harsh/jaeger/plugin/storage/badger/samplingstore/storage_test.go:143
                                                        /home/harsh/jaeger/plugin/storage/badger/samplingstore/storage_test.go:47
                Error:          Not equal:
                                expected: []*model.Throughput{(*model.Throughput)(0xc000400150), (*model.Throughput)(0xc000400180)}
                                actual  : []*model.Throughput(nil)

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,15 +1,2 @@
                                -([]*model.Throughput) (len=2) {
                                - (*model.Throughput)({
                                -  Service: (string) (len=6) "my-svc",
                                -  Operation: (string) (len=2) "op",
                                -  Count: (int64) 0,
                                -  Probabilities: (map[string]struct {}) <nil>
                                - }),
                                - (*model.Throughput)({
                                -  Service: (string) (len=7) "our-svc",
                                -  Operation: (string) (len=3) "op2",
                                -  Count: (int64) 0,
                                -  Probabilities: (map[string]struct {}) <nil>
                                - })
                                -}
                                +([]*model.Throughput) <nil>
```

you can see the timestamp is same 
```
usec:  2024-01-15 02:03:49.674922 +0530 IST
start:  2024-01-15 02:03:49.674922 +0530 IST
```

Reply:
what are `usec` and `start`? who logs them? where are they measured?

Reply:
those are just print statements that i used to print timestamps.

Reply:
```go
func (s *SamplingStore) GetThroughput(start, end time.Time) ([]*model.Throughput, error) {
	var retSlice []*model.Throughput
	prefix := []byte{throughputKeyPrefix}

	err := s.store.View(func(txn *badger.Txn) error {
		opts := badger.DefaultIteratorOptions
		it := txn.NewIterator(opts)
		defer it.Close()

		val := []byte{}
		for it.Seek(prefix); it.ValidForPrefix(prefix); it.Next() {
			item := it.Item()
			k := item.Key()
			startTime := k[1:9]
			val, err := item.ValueCopy(val)
			if err != nil {
				return err
			}
			t, err := initalStartTime(startTime)
			if err != nil {
				return err
			}
			throughputs, err := decodeThroughputValue(val)
			if err != nil {
				return err
			}
			fmt.Println("start: ", start.Truncate(time.Microsecond))
			if t.After(start) && (t.Before(end) || t.Equal(end)) {
				retSlice = append(retSlice, throughputs...)
			}
		}
		return nil
	})
	if err != nil {
		return nil, err
	}

	return retSlice, nil
}
```

```go
func initalStartTime(timeBytes []byte) (time.Time, error) {
	var usec int64

	buf := bytes.NewReader(timeBytes)

	if err := binary.Read(buf, binary.BigEndian, &usec); err != nil {
		return time.Time{}, err
	}
	fmt.Println("usec: ", time.UnixMicro(usec).Truncate(time.Microsecond))
	t := time.UnixMicro(usec)
	return t, nil
}
```

Reply:
These print statements are in unrelated placed, they do not tell me anything. What I would rather see is a print before `if t.After(start) && (t.Before(end) || t.Equal(end))` that outputs `t`, `start`, and `end`, with max precision. Then we can compare their values in successful and failed tests.

Reply:
logs of a successful test.
```
harsh@MSI:~/jaeger/plugin/storage/badger/samplingstore$ go test
badger 2024/01/15 10:55:41 INFO: All 0 tables opened in 0s
badger 2024/01/15 10:55:41 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 10:55:41 INFO: Set nextTxnTs to 0
badger 2024/01/15 10:55:41 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 10:55:41 INFO:
Level 0 [ ]: NumTables: 01. Size: 282 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
badger 2024/01/15 10:55:41 INFO: All 0 tables opened in 0s
badger 2024/01/15 10:55:41 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 10:55:41 INFO: Set nextTxnTs to 0
---------timestamps-----------
t:  2024-01-15 10:55:41.423455 +0530 IST
start:  2024-01-15 10:55:41.423454 +0530 IST
end:  2024-01-15 10:55:51.423454 +0530 IST
badger 2024/01/15 10:55:41 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 10:55:41 INFO:
Level 0 [ ]: NumTables: 01. Size: 282 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
badger 2024/01/15 10:55:41 INFO: All 0 tables opened in 0s
badger 2024/01/15 10:55:41 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 10:55:41 INFO: Set nextTxnTs to 0
badger 2024/01/15 10:55:41 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 10:55:41 INFO:
Level 0 [ ]: NumTables: 01. Size: 278 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
badger 2024/01/15 10:55:41 INFO: All 0 tables opened in 0s
badger 2024/01/15 10:55:41 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 10:55:41 INFO: Set nextTxnTs to 0
badger 2024/01/15 10:55:41 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 10:55:41 INFO:
Level 0 [ ]: NumTables: 01. Size: 332 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
PASS
ok      github.com/jaegertracing/jaeger/plugin/storage/badger/samplingstore     0.268s
```


logs of a failed test
```
badger 2024/01/15 10:56:38 INFO: All 0 tables opened in 0s
badger 2024/01/15 10:56:38 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 10:56:38 INFO: Set nextTxnTs to 0
badger 2024/01/15 10:56:38 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 10:56:38 INFO:
Level 0 [ ]: NumTables: 01. Size: 282 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
badger 2024/01/15 10:56:38 INFO: All 0 tables opened in 0s
badger 2024/01/15 10:56:38 INFO: Discard stats nextEmptySlot: 0
badger 2024/01/15 10:56:38 INFO: Set nextTxnTs to 0
---------timestamps-----------
t:  2024-01-15 10:56:38.23525 +0530 IST
start:  2024-01-15 10:56:38.23525 +0530 IST
end:  2024-01-15 10:56:48.23525 +0530 IST
badger 2024/01/15 10:56:38 INFO: Lifetime L0 stalled for: 0s
badger 2024/01/15 10:56:38 INFO:
Level 0 [ ]: NumTables: 01. Size: 282 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
--- FAIL: TestGetThroughput (0.06s)
    storage_test.go:59:
                Error Trace:    /home/harsh/jaeger/plugin/storage/badger/samplingstore/storage_test.go:59
                                                        /home/harsh/jaeger/plugin/storage/badger/samplingstore/storage_test.go:143
                                                        /home/harsh/jaeger/plugin/storage/badger/samplingstore/storage_test.go:47
                Error:          Not equal:
                                expected: []*model.Throughput{(*model.Throughput)(0xc0002c34a0), (*model.Throughput)(0xc0002c34d0)}
                                actual  : []*model.Throughput(nil)

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,15 +1,2 @@
                                -([]*model.Throughput) (len=2) {
                                - (*model.Throughput)({
                                -  Service: (string) (len=6) "my-svc",
                                -  Operation: (string) (len=2) "op",
                                -  Count: (int64) 0,
                                -  Probabilities: (map[string]struct {}) <nil>
                                - }),
                                - (*model.Throughput)({
                                -  Service: (string) (len=7) "our-svc",
                                -  Operation: (string) (len=3) "op2",
                                -  Count: (int64) 0,
                                -  Probabilities: (map[string]struct {}) <nil>
                                - })
                                -}
                                +([]*model.Throughput) <nil>

                Test:           TestGetThroughput
```